File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/Blockchain.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

public enum Blockchain {
  EHTHEREUM,
  POLYGON,
  NO_CHAIN,
  UNKNOWN_CHAIN;

  public IdentityEntitiesV2.Blockchain toProto() {
    switch (this) {
      case EHTHEREUM:
        return IdentityEntitiesV2.Blockchain.ETHEREUM;
      case POLYGON:
        return IdentityEntitiesV2.Blockchain.POLYGON;
      case NO_CHAIN:
        return IdentityEntitiesV2.Blockchain.NO_CHAIN;
      case UNKNOWN_CHAIN:
        return IdentityEntitiesV2.Blockchain.UNKNOWN_CHAIN;
      default:
        return IdentityEntitiesV2.Blockchain.POLYGON;
    }
  }
}

File: languages/js/src/entity/identity_v2/blockchain.ts
import { Blockchain as BlockchainProto } from "../../bridge/proto/identity_entities_v2";

export enum Blockchain {
  UNRECOGNIZED = -1,
  ETHEREUM = 0,
  POLYGON = 1,
  NO_CHAIN = 2,
  UNKNOWN_CHAIN = 3
}

export namespace Blockchain {
  export function toProto(blockchain: Blockchain): BlockchainProto {
    switch (blockchain) {
      case Blockchain.ETHEREUM:
        return BlockchainProto.ETHEREUM;
      case Blockchain.POLYGON:
        return BlockchainProto.POLYGON;
      case Blockchain.NO_CHAIN:
        return BlockchainProto.NO_CHAIN;
      case Blockchain.UNKNOWN_CHAIN:
        return BlockchainProto.UNKNOWN_CHAIN;
      default:
        return BlockchainProto.UNRECOGNIZED;
    }
  }

  export function fromProto(
    blockchain: BlockchainProto | undefined
  ): Blockchain {
    switch (blockchain) {
      case BlockchainProto.ETHEREUM:
        return Blockchain.ETHEREUM;
      case BlockchainProto.POLYGON:
        return Blockchain.POLYGON;
      case BlockchainProto.NO_CHAIN:
        return Blockchain.NO_CHAIN;
      case BlockchainProto.UNKNOWN_CHAIN:
        return Blockchain.UNKNOWN_CHAIN;
      default:
        return Blockchain.UNRECOGNIZED;
    }
  }
}
